File: apps/docs/src/code/js/basics/fs_read_bytes.js
import fs from 'fs'
import { Sandbox } from 'e2b'

const sandbox = await Sandbox.create({ template: 'base' })

// File bytes will read file's content as bytes
// `fileBytes` as a Uint8Array
const fileBytes = await sandbox.filesystem.readBytes('/etc/hosts') // $HighlightLine

// The output will look similar to this:
// <Buffer 31 32 37 2e 30 2e 30 2e 31 09 6c 6f 63 61 6c 68 6f 73  ...
console.log(fileBytes)

// We can save those bytes to a file locally like this:
fs.writeFileSync('hosts', fileBytes)

await sandbox.close()

File: apps/docs/src/code/python/basics/fs_read_bytes.py
from e2b import Sandbox

sandbox = Sandbox(template="base")

# File bytes will read file's content as bytes
# `file_bytes` as a bytearray
file_bytes = sandbox.filesystem.read_bytes("/etc/hosts")  # $HighlightLine

# The output will look similar to this:
# b'127.0.0.1\tlocalhost\n::1\tlocalhost ip6-localhost ip6-loopback\nfe00::0\tip6-localnet\nff00::0\tip6-mcastprefix\nff02::1\tip6-allnodes\nff02::2\tip6-allrouters\n172.17.0.17\t77c7a543226b\n'
print(file_bytes)

# We can save those bytes to a file locally like this:
with open("./hosts.txt", "wb") as f:
    f.write(file_bytes)

sandbox.close()
